internal/provider/kubernetes/httproute.go

```diff
@@ -31,17 +30,14 @@ const (
 )
 
 type httpRouteReconciler struct {
-	client client.Client
-	log    logr.Logger
-
-	initializeOnce sync.Once
-	resources      *message.ProviderResources
+	client    client.Client
+	log       logr.Logger
+	resources *message.ProviderResources
 }
 
 // newHTTPRouteController creates the httproute controller from mgr. The controller will be pre-configured
 // to watch for HTTPRoute objects across all namespaces.
 func newHTTPRouteController(mgr manager.Manager, cfg *config.Server, resources *message.ProviderResources) error {
-	resources.Initialized.Add(1)
```

Comment:
removing this will break the restart case for xds resources, since it requires the SoTW of all 3 resource types to be able to translate correctly
the easiest thing to do here would be to create a `InitializedXdsIR` and `InitializedInfraIR` variables and set them accordingly

Reply:
@arkodg commit [643dcdf](https://github.com/envoyproxy/gateway/pull/381/commits/643dcdf871c20a7a89690711697db379c0b7f5d4) resolves this feedback. I manually verified that restart still works by using the quickstart guide, restarting the EG deployment, reviewing the EG logs, and verifying connectivity through Envoy.